static/panes/compiler.ts

```diff
@@ -2416,6 +2416,17 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
         this.statusIcon = this.domRoot.find('.status-icon');
 
         this.monacoPlaceholder = this.domRoot.find('.monaco-placeholder');
+
+        $(this.domRoot).on('keydown', event => {
+            if ((event.ctrlKey || event.metaKey) && String.fromCharCode(event.which).toLowerCase() === 's') {
+                event.preventDefault();
+                if (this.assembly.length > 0) {
+                    const texts = this.assembly.map(asm => (asm.text ? asm.text : ''));
+                    const blob = new Blob([texts.join('\n')], {type: 'text/plain;charset=utf-8'});
+                    fileSaver.saveAs(blob, this.id + '.asm');
```

Comment:
It'd be nice if we could provide a more descriptive filename. The editor gives filenames like `Compiler Explorer C++ Editor #1 Code`, we should probably do the same here but "Compiler Output" rather than "Editor" or whatnot :)

Reply:
How about this, `const fileName = this.getLanguageName() + '-' + this.getCompilerName() + '-' + this.id + '.asm';`, e.g. `C-clang default-1.asm`?